### PR TITLE
EDSC-2895: Fix download granule dropdown in table view

### LINKED ISSUE #1227: EDSC-2895: Direct download button not working for multi-file granules in list view
### Description
Some collections consist of granules that are "multi-file" granules. In this case, clicking the "Download single granule data" button should display a dropdown of the available files to download. This is not working when a user is in the table view on the granules panel.

**Reproduction Steps**

1. Search for a collection that has multi-file granules and click on it to load granules (NSIDC-0481 Prod environment)
2. Click on the download icon in the list view and observe that multiple files appear
4. Switch to table view using the toggle in the top-right of the granules panel
5. Click on the download icon in the table view and observe that nothing happens

### Acceptance Criteria

1. Search for a collection that has multi-file granules and click on it to load granules (NSIDC-0481 in Prod)
2. Click on the download icon in the list view and observe that multiple files appear
3. Confirm that clicking on those links downloads the intended file
4. Switch to table view using the toggle at the top of the granules panel
5. Click on the download icon in the table view and observe that multiple files appear
6. Confirm that clicking on those links downloads the intended file

### PR DESCRIPTION
**Pull request by saurabhdaware**
# Overview

### What is the feature?

Fixes #1227 

### What is the Solution?

~Not 100% sure of the possible cause but it was possibly because rowClick event of table triggering after the onClick of download button so it was immediately removing the dropdown menu.~

Edit: Changed implementation to use `e.stopPropogation()`

### What areas of the application does this impact?

~It will impact the parts which has the same `onRowClick` function reference that I've made changes into. 
It checks if it has the `granule-results-data-links-button__button` class to check if it is the download button element.~

# Testing

### Reproduction steps

- **Environment for testing:**
- **Collection to test with:**

1. Search for a collection that has multi-file granules and click on it to load granules (NSIDC-0481 in 
2. Click on the download icon in the list view and observe that multiple files appear
3.  Confirm that clicking on those links downloads the intended file
4. Switch to table view using the toggle at the top of the granules panel
5. Click on the download icon in the table view and observe that multiple files appear
6. Confirm that clicking on those links downloads the intended file

### Attachments

Please include relevant screenshots or files that would be helpful in reviewing and verifying this change.

# Checklist

- [ ] I have added automated tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings

### CODE REVIEW

**Comment by saurabhdaware** [comment]
Need to do a tiny change. I'll mark it for review after that.

**Comment by abbottry** [comment]
I think that the change necessary here is to just add `e.stopPropagation()` within `handleClick` for the `CustomDataLinksToggle` component. If that does the trick please ensure that the z-index for the rows look correct, there may be a change necessary.

**Comment by saurabhdaware** [comment]
Yup right! `e.stopPropagation()` worked. Also, didn't need to change z-index 
screenshot of collection page in table view that shows dropdown items

**Comment by mreese84** [comment]
@saurabhdaware good deal. However, notice in your screenshot that there is a green line that goes over the top of the dropdown list. That should be under the dropdown.

Also, a reminder to make sure you have a test that covers this bugfix. Thanks!

**Comment by saurabhdaware** [comment]
Yup, I'll make those changes and add tests. Thank you!

**Comment by saurabhdaware** [comment]
I've used `React.createPortal` to render the menu outside of the table row. This does solve the z-axis issue.
image

I'll write tests now. Let me know if this is fine. Thanks!

File: static/src/js/components/GranuleResults/__tests__/GranuleResultsDataLinksButton.test.js [hunk 1]
```diff
@@ -76,6 +77,9 @@ describe('GranuleResultsDataLinksButton component', () => {
 
   describe('with multiple granule links', () => {
     test('renders the correct element', () => {
+      // Mocks createPortal method of ReactDOM (https://stackoverflow.com/a/60953708/8116576)
+      ReactDOM.createPortal = jest.fn(modal => modal);
```
Comment (trevorlang) [inline static/src/js/components/GranuleResults/__tests__/GranuleResultsDataLinksButton.test.js:81]: It's strange to use `modal` in this case. I would prefer to see this named `dropdown` or something more specific to this implementation.

**Review by trevorlang — CHANGES_REQUESTED** [review]
@saurabhdaware, this is looking awesome. I have one small suggestion. Once that's patched up, this should be good to go.

File: static/src/js/components/GranuleResults/__tests__/GranuleResultsDataLinksButton.test.js (hunk 1 above)
Comment (saurabhdaware) [reply static/src/js/components/GranuleResults/__tests__/GranuleResultsDataLinksButton.test.js:81]: Yes, right! I'll make that change. 

Also, we will have to write a test that checks if dropdown opens when it is clicked from table view. (This test only checks if it opens when there are more links). I'll add that test as well then we can merge I guess.

Comment (abbottry) [reply static/src/js/components/GranuleResults/__tests__/GranuleResultsDataLinksButton.test.js:81]: Yes, that is correct. And once this is done, it'll be ready to merge!

**Comment by saurabhdaware** [comment]
Hi, I've added test for `CustomDataLinksToggle` function that checks if `e.stopPropagation` and other required methods are fired on click. Let me know if you want any other changes. Thanks!

File: static/src/js/components/GranuleResults/__tests__/GranuleResultsDataLinksButton.test.js (hunk 1 above)
Comment (saurabhdaware) [reply static/src/js/components/GranuleResults/__tests__/GranuleResultsDataLinksButton.test.js]: Fixed this and added a test for CustomDataLinksToggle component

**Review by macrouch — APPROVED** [review]

**Review by trevorlang — APPROVED** [review]
Looks good! Thanks!

**Comment by mreese84** [comment]
@saurabhdaware thank you for your contribution! This fix will go out in our 1.130.x release, which is scheduled for deployment to our Prod environment no later than 10/28/20. Thanks!

[search.earthdata.nasa.gov](https://search.earthdata.nasa.gov)

**Pull request merged**